### PR TITLE
#165853243 Fix User Registration and Login on Swagger

### PR DESCRIPTION
**Pull request by Benkimeric**
#### What does this PR do?
- fix user registration and login on swagger to allow nesting
#### Description of Task to be completed?
- replace `django-rest-swagger` with `drf-yasg`
- overide default swagger properties in registration and login to allow nesting
- include class meta in follows to enable rendering in `drf-yasg`
- refactor code to remove and include serializers for rendering in the swagger UI

### How should you manually test this?

- [x] **Clone the repository**

        $ git clone https://github.com/andela/ah-the-jedi-backend.git

- [x] **Switch to testing branch**

       $ git checkout ft-like-dislike-articles-165305269

- [x] **Switch to the parent directory**

       $ cd ah-the-jedi-backend/

 

- [x] **Create and activate the virtual environment**

      $  virtualenv -p python .venv

      $ source .venv/bin/activate

- [x] **Install project dependencies**

       $ pip install -r api/requirements.txt

- [x] **Run the database migrations**

       $ python api/manage.py migrate

### Running the application

      $ python api/manage.py runserver

### Go to the URL to view the documentation and interact with any endpoint
```<your-localhost>/```

#### Any background context you want to provide?
None
#### What are the relevant pivotal tracker stories?
[#165853243](https://www.pivotaltracker.com/story/show/165853243)

### Swagger view for Registration
Screenshot 2019-05-15 at 13 16 47

### Swagger view for Login
Screenshot 2019-05-15 at 13 17 52

#### Questions:
Any feedback?

### CODE REVIEW

**Review by jkamz — APPROVED** [review]
Looks good. Well done.

**Review by Siffersari — APPROVED** [review]
LGTM

**Review by ElishaMisoi — APPROVED** [review]
Good job 👍🏽

File: authors/apps/authentication/views.py [hunk 1]
```diff
@@ -78,6 +101,21 @@ class LoginAPIView(GenericAPIView):
     renderer_classes = (UserJSONRenderer,)
     serializer_class = LoginSerializer
 
+    @swagger_auto_schema(
```
Comment (katherine95) [inline authors/apps/authentication/views.py:104]: Good work in implementing nesting of fields in the documentation. LGTM

**Review by katherine95 — APPROVED** [review]

**Review by AlvinMugambi — APPROVED** [review]

**Review by rnjane — APPROVED** [review]

**Pull request merged**